cli/src/main/java/com/devonfw/tools/ide/tool/LocalToolCommandlet.java

```diff
@@ -83,14 +83,16 @@ protected boolean doInstall(boolean silent) {
         step.success();
         return false;
       }
-      // we need to link the version or update the link.
-      Path toolPath = getToolPath();
-      FileAccess fileAccess = this.context.getFileAccess();
-      if (Files.exists(toolPath)) {
-        fileAccess.backup(toolPath);
+      if (!isIgnoreSoftwareRepo()) {
+        // we need to link the version or update the link.
+        Path toolPath = getToolPath();
+        FileAccess fileAccess = this.context.getFileAccess();
+        if (Files.exists(toolPath)) {
+          fileAccess.backup(toolPath);
+        }
+        fileAccess.mkdirs(toolPath.getParent());
```

Comment:
Isn't that duplicated now? See lines 195-198.

Reply:
`ToolPath` doesnt have the same value in both methods, in `doInstall` it's the software folder, and in `installTool` it's either the software folder or software repo. So in case we want to change the version of a tool we need to back up the tool in software folder, I couldn't find a better way to avoid this redundancy.